### PR TITLE
DDF-1528: TestCswSource test cases failing

### PR DESCRIPTION
**Pull request by lessarderic**
Changed TestCswSourceBase.getGetRecordsTypeAsXml() to use the same CswJAXBElementProvider class CswSource is using and added the proper namespace on the typeNames attribute value Record to make the tests pass again.
Updated assertions that were inverting expected vs. actual.
Removed unused namespaces from expected requests, and switched default namespace to something more reasonable.
Disabled XMLUnit attribute order and comment validation.
Added more detailed output if the assertion is about to fail to help in case we run into other build failures.
Fixed compilation warnings.

@kcwire, @pklinef, @coyotesqrl please review.

[Review on Reviewable](https://reviewable.io/reviews/codice/ddf/211)

### CODE REVIEW

**Comment by coyotesqrl** [comment]
Looks good. I especially like the additional, more verbose, logging.


---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks pending.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/211)

**Comment by kcwire** [comment]
:+1:

**Comment by coyotesqrl** [comment]
:+1: 


---

Review status: 0 of 3 files reviewed at latest revision, all discussions resolved, some commit checks failed.

---



Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/codice/ddf/211)

File: catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java [hunk 1]
```diff
@@ -209,9 +210,16 @@ public void testPropertyIsLikeQuery()
         }
         GetRecordsType getRecordsType = captor.getValue();
 
-        String xml = getGetRecordsTypeAsXml(getRecordsType, CswConstants.VERSION_2_0_2);
-        LOGGER.debug(xml);
-        assertXMLEqual(xml, getRecordsControlXml202);
+        String xml = getGetRecordsTypeAsXml(getRecordsType);
```
Comment (pklinef) [inline catalog/spatial/csw/spatial-csw-source/src/test/java/org/codice/ddf/spatial/ogc/csw/catalog/source/TestCswSource.java]: Looks like this block of code is copied multiple times.  Any chance of extracting a method?

**Pull request merged**